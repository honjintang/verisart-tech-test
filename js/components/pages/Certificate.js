import React from 'react'
import { withRouter } from 'react-router'

const Certificate = () => (
    <div>Certificate</div>
)

export default withRouter(Certificate)