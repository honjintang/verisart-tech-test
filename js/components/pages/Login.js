import React from 'react'
import { connect } from 'react-redux'
import TextInput from '~/components/general/TextInput'
import Button from '~/components/general/Button'
import { login } from '~/actions'
import fb from '../../../assets/images/fb.svg'
import google from '../../../assets/images/google.svg'
import { getEmail, getPassword } from '~/reducers/auth'

const mapStateToProps = state => ({
    email: getEmail(state),
    password: getPassword(state),
})

const mapDispatchToProps = dispatch => ({
    _login: ({ email, password }) => dispatch(login({ email, password })),
  })

const Login = ({ _login, email, password }) => (
    <div>
        <div className="loginTitle">Login</div>
        <div className="loginBox">
            <div className="loginFormWrapper">
                <span></span>
                <div className="loginFormContent">
                    <div className="loginWithEmail">
                        <TextInput type="email" name="email" placeholder="Email" value={email} />
                        <div className="loginSpacing">
                            <TextInput type="password" name="password" placeholder="Enter Password" value={password}  />
                        </div>
                        <div className="loginFooter">
                            <Button text="Login" onClickFn={() => _login({ email, password })} />
                        </div>
                        <div className="forgotPasswordWrapper">
                            <a className="forgotPassword" href="/forgot-password">Forgot Password</a>
                        </div>
                        <div className="socialWrapper">
                            <div className="loginWithSocial">
                                <div className="loginWithSocialWrapper">
                                    <div className="loginSocialButton">
                                        <div className="loginSocialIcon">
                                            <img className="socialIcon" src={fb} width="22.5px" height="22.5px"/>
                                        </div>
                                        <div className="loginSocialLabel">
                                            Login with Facebook
                                        </div>
                                    </div>
                                </div>
                            </div>
                            <div className="loginWithSocial">
                                <div className="loginWithSocialWrapper">
                                    <div className="loginSocialButton">
                                        <div className="loginSocialIcon">
                                            <img className="socialIcon" src={google} width="36px" height="36px" />
                                        </div>
                                        <div className="loginSocialLabel">
                                            Login with Google
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
)

export default connect(
    mapStateToProps,
    mapDispatchToProps
  )(Login)