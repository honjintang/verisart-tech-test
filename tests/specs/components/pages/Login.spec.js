import React from 'react'
import { shallow } from 'enzyme'
import { Login } from '~/components/pages/Login'

const mockLogin = jest.fn()

describe('components/display/Login', () => {
    it('matches snapshot', () => {
        const component = shallow(<Login _login={mockLogin} email='email' password='password' />)
        expect(component).toMatchSnapshot()
    })
})