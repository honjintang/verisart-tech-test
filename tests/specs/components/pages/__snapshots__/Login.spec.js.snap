// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/Login matches snapshot 1`] = `
<div>
  <div
    className="loginTitle"
  >
    Login
  </div>
  <div
    className="loginBox"
  >
    <div
      className="loginFormWrapper"
    >
      <span />
      <div
        className="loginFormContent"
      >
        <div
          className="loginWithEmail"
        >
          <Connect(TextInput)
            name="email"
            placeholder="Email"
            type="email"
            value="email"
          />
          <div
            className="loginSpacing"
          >
            <Connect(TextInput)
              name="password"
              placeholder="Enter Password"
              type="password"
              value="password"
            />
          </div>
          <div
            className="loginFooter"
          >
            <Button
              onClickFn={[Function]}
              text="Login"
            />
          </div>
          <div
            className="forgotPasswordWrapper"
          >
            <a
              className="forgotPassword"
              href="/forgot-password"
            >
              Forgot Password
            </a>
          </div>
          <div
            className="socialWrapper"
          >
            <div
              className="loginWithSocial"
            >
              <div
                className="loginWithSocialWrapper"
              >
                <div
                  className="loginSocialButton"
                >
                  <div
                    className="loginSocialIcon"
                  >
                    <img
                      className="socialIcon"
                      height="22.5px"
                      src="test-file-stub"
                      width="22.5px"
                    />
                  </div>
                  <div
                    className="loginSocialLabel"
                  >
                    Login with Facebook
                  </div>
                </div>
              </div>
            </div>
            <div
              className="loginWithSocial"
            >
              <div
                className="loginWithSocialWrapper"
              >
                <div
                  className="loginSocialButton"
                >
                  <div
                    className="loginSocialIcon"
                  >
                    <img
                      className="socialIcon"
                      height="36px"
                      src="test-file-stub"
                      width="36px"
                    />
                  </div>
                  <div
                    className="loginSocialLabel"
                  >
                    Login with Google
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
